dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/NettyServiceGateway.scala

```diff
@@ -119,10 +119,10 @@ class NettyServiceGateway(coordinatedShutdown: CoordinatedShutdown, config: Serv
             val path = URI.create(request.uri).getPath
 
             implicit val ec = ctx.executor
-            (registry ? Route(request.method.name, path)).mapTo[RouteResult].map {
+            (registry ? Route(request.method.name, path, None)).mapTo[RouteResult].map {
               case Found(serviceAddress) =>
                 log.debug("Request is to be routed to {}, getting connection from pool", serviceAddress)
-                val pool = poolMap.get(serviceAddress)
+                val pool = poolMap.get(new InetSocketAddress(serviceAddress.getHost, serviceAddress.getPort))
```

Comment:
Note that `new InetSocketAddress` is a blocking operation (not sure whether or not that's a problem here).

> An attempt will be made to resolve the hostname into an InetAddress. If that attempt fails, the address will be flagged as unresolved.

https://docs.oracle.com/javase/7/docs/api/java/net/InetSocketAddress.html#InetSocketAddress(java.lang.String,%20int)

Reply:
Good call. In the previous implementation the `InetSocketAddress` was only created when registering a service on the registry and after that moment the instance was reused.